### PR TITLE
Add unlabeled abacus STRU read/dump interface

### PR DESCRIPTION
**Pull request by Liu-RX**

### CODE REVIEW

File: dpdata/plugins/abacus.py [hunk 1]
```diff
@@ -1,7 +1,47 @@
+from ast import For
```
Comment (njzjz) [inline dpdata/plugins/abacus.py:1]: typo?

File: dpdata/plugins/abacus.py [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
 import dpdata.abacus.scf
 import dpdata.abacus.md
 from dpdata.format import Format
 
+@Format.register("abacus/stru")
+@Format.register("stru")
+class AbacusSTRUFormat(Format):
+    def from_system(self, file_name, **kwargs):
+        return dpdata.abacus.scf.get_frame_from_stru(file_name)
```
Comment (njzjz) [inline dpdata/plugins/abacus.py:10]: a blank line should be after this line

File: dpdata/plugins/abacus.py [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+            List of atomic masses
+        numerical_descriptor: str, optional
+            numerical descriptor file
+        """
+        pp_file = None
+        numerical_orbital = None
+        mass = None
+        numerical_descriptor = None
+        if 'pp_file' in kwargs:
+            pp_file = kwargs['pp_file']
```
Comment (njzjz) [inline dpdata/plugins/abacus.py:35]: can be written as `pp_file = kwargs.get('pp_file')`

File: tests/test_abacus_pw_scf.py [hunk 4]
```diff
@@ -88,15 +89,15 @@ def test_force(self) :
...
-                self.assertAlmostEqual(self.system_ch4.data['virials'][0][ii][jj], virial[ii][jj], places = 3)
-        fp.close()
+        with open('abacus.scf/ch4_virial') as fp:
+            virial = []
+            for ii in fp :
+                virial.append([float(jj) for jj in ii.split()])
+            virial = np.array(virial)
+            for ii in range(virial.shape[0]) :
+                for jj in range(virial.shape[1]) :
+                    self.assertAlmostEqual(self.system_ch4.data['virials'][0][ii][jj], virial[ii][jj], places = 3)
```
Comment (njzjz) [inline tests/test_abacus_pw_scf.py:99]: `np.testing.assert_almost_equal` can be used to compare numpy arrays.

File: dpdata/plugins/abacus.py (hunk 1 above)
Comment (Liu-RX) [reply dpdata/plugins/abacus.py]: Yes! Thanks for the careful inspection!

File: dpdata/plugins/abacus.py (hunk 2 above)
Comment (Liu-RX) [reply dpdata/plugins/abacus.py:9]: Got that. I have already added the blank line.

File: dpdata/plugins/abacus.py (hunk 3 above)
Comment (Liu-RX) [reply dpdata/plugins/abacus.py]: I have changed all of the cases into `kwargs.get()`.

File: tests/test_abacus_pw_scf.py (hunk 4 above)
Comment (Liu-RX) [reply tests/test_abacus_pw_scf.py]: Thank you for the most helpful suggestion! I have changed all possible cases in both `test_abacus_pw_scf.py` and `test_abacus_md.py` into `np.testing.assert_almost_equal`.

File: dpdata/plugins/abacus.py (hunk 3 above)
Comment (njzjz) [reply dpdata/plugins/abacus.py]: When you write as this case, line 34 can be removed.

**Review by njzjz — APPROVED** [review]

File: dpdata/plugins/abacus.py (hunk 3 above)
Comment (wanghan-iapcm) [reply dpdata/plugins/abacus.py]: @Liu-RX plz remove the `if` clauses

Comment (Liu-RX) [reply dpdata/plugins/abacus.py]: I have removed the `if` clauses and the `pp_file = None` lines.

**Pull request merged**